### ISSUE TITLE
MiniMax-M3: forward does not declare logits_to_keep, so fused losses are silently replaced

### ISSUE THREAD

**Issue by coderaBruce** [NVIDIA-NeMo/Automodel#3510]
### Describe the bug

`MiniMaxM3SparseForConditionalGeneration.forward` does not declare `logits_to_keep`; it absorbs it into `**kwargs`. Because `_supports_logits_to_keep` inspects the forward signature, the check returns `False` and the recipe silently replaces the configured loss:

[https://github.com/NVIDIA-NeMo/Automodel/blob/main/nemo_automodel/recipes/vlm/finetune.py#L554-L556](<https://github.com/NVIDIA-NeMo/Automodel/blob/main/nemo_automodel/recipes/vlm/finetune.py#L554-L556>)

```python
if not _supports_logits_to_keep(model) and not isinstance(self.loss_fn, MaskedCrossEntropy):
    logger.warning("logits_to_keep not found in model.forward. Using MaskedCrossEntropy instead.")
    self.loss_fn = MaskedCrossEntropy()
```

Two consequences:

1. **The full logits tensor is materialised.** M3's vocab is 200,064, so `[tokens, 200064]` is large at long context — at `cp=8` with a 200,704-token sequence that is 25,088 local tokens x 200,064 = ~10 GB in bf16.
2. **The configured** `loss_fn`**'s settings are dropped with it.** The replacement is a *fresh* `MaskedCrossEntropy()`, so a recipe that sets `fp32_upcast: false` loses it and the same tensor is upcast to fp32, roughly doubling the cost.

Callers already pass the kwarg unconditionally ([`finetune.py#L930`](<https://github.com/NVIDIA-NeMo/Automodel/blob/main/nemo_automodel/recipes/vlm/finetune.py#L930>), [`#L1179`](<https://github.com/NVIDIA-NeMo/Automodel/blob/main/nemo_automodel/recipes/vlm/finetune.py#L1179>)), so today it is accepted and then ignored — `**kwargs` swallows it and nothing in the forward reads it.

### Why this looks like an oversight rather than a design choice

The stricter check is correct for this kwarg. `_supports_seq_lens`, immediately below it in the same file, *does* accept `**kwargs` — because merely accepting `seq_lens` is harmless. Accepting `logits_to_keep` is not equivalent to honouring it: the model has to return hidden states instead of logits, so requiring an explicit parameter is a reasonable proxy for "this model implements the contract".

Every other custom model in `components/models` declares it — `minimax_m2`, `glm_moe_dsa`, `deepseek_v4`, `gemma4_moe`, `qwen3_moe`, `gpt_oss`, `llama`, `qwen3_next`, `glm4_moe`, `step3p7`. MiniMax-M3 is the only one that does not.

### Steps/Code to reproduce bug

Configure any M3 recipe with a fused loss, e.g.:

```yaml
loss_fn:
  _target_: nemo_automodel.components.loss.linear_ce.FusedLinearCrossEntropy
```

The run logs `logits_to_keep not found in model.forward. Using MaskedCrossEntropy instead.` and proceeds with full fp32 logits. Alternatively, without launching a job:

```python
import inspect
from nemo_automodel.components.models.minimax_m3_vl.model import (
    MiniMaxM3SparseForConditionalGeneration,
)

sig = inspect.signature(MiniMaxM3SparseForConditionalGeneration.forward)
print("logits_to_keep" in sig.parameters)  # False
```

### Expected behavior

`FusedLinearCrossEntropy` (and `ChunkedCrossEntropy`) should be usable with M3: `forward` declares `logits_to_keep` and, when it is set, returns `{"hidden_states": hidden}` without applying `lm_head`, so the logits tensor is never built and the configured loss survives.

### Environment overview

* Container: `nvcr.io/nvidia/nemo-automodel:26.06`
* Automodel: reproduced against `main` (`6551ee4`) and `f59b2425`
* Hardware: 8xB300, single node
* Model: `MiniMaxAI/MiniMax-M3`, `num_mtp_modules: 0`

### Additional context

Found while fine-tuning M3 with LoRA + `cp=8`. Separately reported: NVIDIA-NeMo/Automodel#3507 (dense layers 0-2 losing context parallelism at `ep_size > 1`).

`MiniMaxM3SparseForCausalLM` has the same gap, but it is not in the architecture registry and its `thd` branch reshapes the return value, so a fix there would need its own validation.

A PR with the minimal fix follows.
